### PR TITLE
Ability to run SG tests on amazon AMI and Ubuntu and sanity cleanup

### PR DESCRIPTION
**Pull request by raghusarangapani**
#### Fixes #.

- [x] Ran `flake8`
- [x] Ran `run_repo_tests.sh`

#### Changes proposed in this pull request:
- Adds ability to run SG tests against Amazon AMI and Ubuntu
- Cleaned up sanity tags - 57 sanity tests.
- This does not provide provisioning capability for Amazon AMI and Ubuntu VMs.

### CODE REVIEW

File: libraries/provision/install_sync_gateway.py [hunk 1]
```diff
@@ -23,24 +23,6 @@ def __init__(self,
...
-            "1.3.1.2",
-            "1.4.0",
-            "1.4",
-            "1.4.0.1",
-            "1.4.0.2",
-            "1.4.1",
-            "1.4.1.1",
-            "1.4.2",
-            "1.5.0"
-        ]
```
Comment (sridevi-15) [inline libraries/provision/install_sync_gateway.py]: reason for removing self._valid_versions?

File: testsuites/syncgateway/functional/tests/test_xattrs.py [hunk 2]
```diff
@@ -512,7 +509,6 @@ def test_offline_processing_of_external_updates(params_from_base_test_setup, sg_
     sg_client.verify_docs_in_changes(url=sg_url, db=sg_db, expected_docs=docs_to_verify_in_changes, auth=seth_auth)
```
Comment (sridevi-15) [inline testsuites/syncgateway/functional/tests/test_xattrs.py]: verificy docs in changes is good to be sanity. Is any sanity test covering changes?

File: testsuites/syncgateway/functional/tests/test_webhooks.py [hunk 3]
```diff
@@ -86,7 +86,6 @@ def test_webhooks(params_from_base_test_setup, sg_conf_name, num_users, num_chan
     assert expected_events == received_events
 
 
-@pytest.mark.sanity
```
Comment (sridevi-15) [inline testsuites/syncgateway/functional/tests/test_webhooks.py]: One webhooks test in sanity will be good

File: libraries/provision/install_sync_gateway.py (hunk 1 above)
Comment (raghusarangapani) [reply libraries/provision/install_sync_gateway.py]: We are not using it anymore. We don't validate the version before installing, we do it after. We used to do it, not anymore.

File: testsuites/syncgateway/functional/tests/test_webhooks.py (hunk 3 above)
Comment (raghusarangapani) [reply testsuites/syncgateway/functional/tests/test_webhooks.py]: There is , test_webhooks

File: testsuites/syncgateway/functional/tests/test_xattrs.py (hunk 2 above)
Comment (raghusarangapani) [reply testsuites/syncgateway/functional/tests/test_xattrs.py]: Marked this test case as sanity

**Pull request merged**